sumcheck/src/prover_helper/arbitrary_gate.rs

```diff
@@ -0,0 +1,187 @@
+                    // Evaluate P({0, ..., |g(r)|})
+
+                    // TODO(#28): Optimize
+                    // Tricks can be used here for low order bits {0,1} but general premise is a running sum for each
+                    // of the m terms in the Dense multilinear polynomials. Formula is:
+                    // half = | D_{n-1} | / 2
+                    // D_n(index, r) = D_{n-1}[half + index] + r * (D_{n-1}[half + index] - D_{n-1}[index])
+
+                    // eval 0: bound_func is A(low)
+                    let params_zero: Vec<F> =
+                        polys.iter().map(|poly| poly.coeffs[poly_term_i]).collect();
+                    accum[0] += comb_func(&params_zero);
+
+                    // TODO(#28): Can be computed from prev_round_claim - eval_point_0
+                    let params_one: Vec<F> = polys
+                        .iter()
+                        .map(|poly| poly.coeffs[mle_half + poly_term_i])
+                        .collect();
+                    accum[1] += comb_func(&params_one);
+
+                    // D_n(index, r) = D_{n-1}[half + index] + r * (D_{n-1}[half + index] - D_{n-1}[index])
+                    // D_n(index, 0) = D_{n-1}[LOW]
+                    // D_n(index, 1) = D_{n-1}[HIGH]
+                    // D_n(index, 2) = D_{n-1}[HIGH] + (D_{n-1}[HIGH] - D_{n-1}[LOW])
+                    // D_n(index, 3) = D_{n-1}[HIGH] + (D_{n-1}[HIGH] - D_{n-1}[LOW]) + (D_{n-1}[HIGH] - D_{n-1}[LOW])
+                    // ...
+                    let mut existing_term = params_one;
+                    for eval_i in accum.iter_mut().take(combined_degree + 1).skip(2) {
+                        let mut poly_evals = vec![F::zero(); polys.len()];
+                        for poly_i in 0..polys.len() {
+                            let poly = &polys[poly_i];
+                            poly_evals[poly_i] = existing_term[poly_i]
+                                + poly.coeffs[mle_half + poly_term_i]
+                                - poly.coeffs[poly_term_i];
+                        }
+
+                        *eval_i += comb_func(&poly_evals);
+                        existing_term = poly_evals;
+                    }
+                    accum
+                })
+                .collect();
+
+            // Vector storing evaluations of combined polynomials g(x) = P_0(x) * ... P_{num_polys} (x)
+            // for points {0, ..., |g(x)|}
+            let uni_poly_eval_points: Vec<F> = (0..combined_degree + 1)
+                .map(|poly_i| accum.iter().map(|mle| mle[poly_i]).sum())
+                .collect();
+
+            let round_uni_poly = UnivariatePolynomial::from_evals(&uni_poly_eval_points);
+
+            // append the prover's message to the transcript
+            // round_uni_poly.
+            // transcript.append_field_element(f);
+            println!("round_uni_poly: {:?}", round_uni_poly.coefficients.len());
```

Comment:
I guess you want to say something related to the degree of the univariate poly in the round of response in sumcheck?